### PR TITLE
[CFM-1178] Updates global contact form

### PR DESCRIPTION
**Pull request by DavidHernandez**
Issue: https://issuetracker.amplexor.com/jira/browse/CFM-1178

### Merge https://github.com/capacity4dev/capacity4more/pull/1262 first

Screenshots:
## Anonymous
contact-anon
### Desktop
contact-desktop
### Tablet
contact-tablet
### Mobile
contact-mobile

### CODE REVIEW

**Comment by DavidHernandez** [comment]
### E-mail being sent
mail-example

**Comment by ordavidil** [comment]
@DavidHernandez 
did you check it for anonymous as well ?

can you please add a test for anonymous users ?

File: project/profiles/capacity4more/behat/features/contact.feature [hunk 1]
```diff
@@ -16,3 +16,26 @@ Feature: Contact
     And I fill in "E-mail addresses" with "info@example.com, other@example.com"
     And I press "Save configuration"
     Then I should see "The configuration options have been saved."
+
+  @api
+  Scenario: Check contact page as anonymous
+    Given I am an anonymous user
+    When I go to "contact"
+    Then I should see an enabled "#edit-name" element
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/contact.feature]: Please change the tests in a way of a real scenario when user try to submit the form

File: project/profiles/capacity4more/behat/features/bootstrap/FeatureContext.php [hunk 2]
```diff
@@ -12,6 +12,7 @@
 // Split FeatureContext in smaller chunks.
 require __DIR__ . '/FeatureContext/Activity.php';
 require __DIR__ . '/FeatureContext/Article.php';
+require __DIR__ . '/FeatureContext/Contact.php';
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/bootstrap/FeatureContext.php]: @DavidHernandez 
I think we no longer need this file.

File: project/profiles/capacity4more/modules/c4m/features/c4m_contact/c4m_contact.module [hunk 3]
```diff
@@ -45,6 +67,23 @@ function c4m_contact_form_contact_site_form_alter(&$form, &$form_state, $form_id
...
+    global $user;
+    $user = user_load($user->uid);
+
+    $user_name = $user->name;
+    // Prevents errors if using admin user, who has no first name set.
+    if (!empty($user->c4m_first_name)) {
+      $user_name = $user->c4m_first_name[LANGUAGE_NONE][0]['value'];
+    }
+
+    $form['name']['#default_value'] = $user_name;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_contact/c4m_contact.module]: use `c4m_user_full_name()`

File: project/profiles/capacity4more/modules/c4m/features/c4m_contact/c4m_contact.module [hunk 4]
```diff
@@ -45,6 +67,23 @@ function c4m_contact_form_contact_site_form_alter(&$form, &$form_state, $form_id
     return;
   }
 
+  if (!user_is_anonymous()) {
+    global $user;
+    $user = user_load($user->uid);
+
+    $user_name = $user->name;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_contact/c4m_contact.module]: remove this logic for the admin (user 1)

File: project/profiles/capacity4more/behat/features/contact.feature [hunk 5]
```diff
@@ -16,3 +16,21 @@ Feature: Contact
...
+    And I should see "Your name field is required."
+    And I should see "Your e-mail address field is required."
+
+  @api
+  Scenario: Check contact page as logged in user
+    Given I am logged in as user "isaacnewton"
+    When I go to "contact"
+    And I press "Send message"
+    Then I should see "Subject field is required"
+    And I should see "Message field is required."
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/contact.feature]: @DavidHernandez 
please add the other two lines as well in here but the other way:
```gherkin
And I should not see "Your name field is required."
And I should not see "Your e-mail address field is required."
```

**Comment by ordavidil** [comment]
@DavidHernandez 
Add one more test to check successful submit.

**Pull request merged**